packages/wallet/src/mintbase-wallet.ts

```diff
@@ -0,0 +1,241 @@
+  callback,
+  networkId,
+}) => {
+  const setupWalletState = async () => {
+    const { connect, WalletConnection } = nearAPI;
+
+    const connectionConfig = {
+      networkId: networkId,
+      keyStore: new BrowserLocalStorageKeyStore(),
+      nodeUrl: 'https://rpc.testnet.near.org',
+      walletUrl: walletUrl,
+      headers: {},
+    };
+
+    const searchParams = new URL(window.location.href);
+
+    const acc = searchParams.searchParams.get('account_id');
+
+    if (acc && acc?.length > 0) {
+      localStorage.setItem('mintbase-wallet_callback_url', callback);
+
+      localStorage.setItem(
+        'mintbase-wallet_wallet_auth_key',
+        JSON.stringify({
+          accountId: acc as string,
+          allKeys: [],
+        }),
+      );
+    }
+
+    const nearConnection = await connect(connectionConfig);
+
+    const wallet = new WalletConnection(nearConnection, 'mintbase-wallet');
+
+    return {
+      wallet,
+    };
+  };
+
+  const state = await setupWalletState();
+
+  let activeAccountId: string;
+
+  const getAccountId = () => activeAccountId;
+
+  const isSignedIn = async () => !!activeAccountId;
+
+  const signIn = async () => {
+    const existingAccounts = await getAccounts();
+
+    const origin = window.location.origin;
+
+    if (existingAccounts.length) {
+      return existingAccounts;
+    }
+
+    await state.wallet.requestSignIn({
+      methodNames: [],
+      successUrl: successUrl ?? origin,
+      failureUrl: failureUrl ?? origin,
+    });
+
+    return getAccounts();
+  };
+
+  const signOut = async () => {
+    // if (activeAccountId === undefined || activeAccountId === null) {
+    //   throw new Error("Wallet is already signed out");
+    // }
+
+    window.localStorage.removeItem('mintbase-wallet:account-data');
+
+    if (state.wallet.isSignedIn()) {
+      state.wallet.signOut();
+    }
+  };
+
+  const assertValidSigner = (signerId: string) => {
+    if (signerId && signerId !== state.wallet.getAccountId()) {
+      throw new Error(
+        `Cannot sign transactions for ${signerId} while signed in as ${activeAccountId}`,
+      );
+    }
+  };
+
+  const signAndSendTransactions = async ({
+    transactions,
+    callbackUrl,
+  }: {
+    transactions: Array<Transaction>;
+    callbackUrl: string;
+  }): Promise<void> => {
+    // throw new Error(
+    //   "Mintbase Wallet does not support signing and sending multiple transactions."
+    // );
+
+    for (const { signerId } of transactions) {
+      assertValidSigner(signerId);
+    }
+    const stringifiedParam = JSON.stringify(transactions);
+
+    const urlParam = encodeURIComponent(stringifiedParam);
+    //const currentUrl = new URL(window.location.href);
+    const newUrl = new URL(`${walletUrl}/sign-transaction`);
+    newUrl.searchParams.set('transactions_data', urlParam);
+    newUrl.searchParams.set('callback_url', callbackUrl);
+
+    window.location.assign(newUrl.toString());
+    return;
+  };
+
+  const signAndSendTransaction = ({
```

Comment:
The input and output of this function signature looks like it might warrant having its own type declaration. Like TxData?

It could be destructured inside the function if need be.